### ISSUE TITLE
FMI coverage regressions in master

### ISSUE THREAD

**Issue by adrpo** [OpenModelica/OpenModelica#8641]
### Description

See: https://libraries.openmodelica.org/branches/overview-fmi.html

BuildingSystems
![image](https://user-images.githubusercontent.com/4192979/156883305-aee4d2b0-1ff8-49e5-b45a-3cae36c62925.png)

Buildings_maint.7.0.x
![image](https://user-images.githubusercontent.com/4192979/156883336-3f706685-1180-4310-a581-d5e3ee5d7683.png)

As far as I can tell some of the failures are with the same error:
https://libraries.openmodelica.org/branches/master-fmi/Buildings_maint.7.0.x/files/Buildings_maint.7.0.x_Buildings.Applications.DataCenters.ChillerCooled.Equipment.Validation.NonIntegrated.err
Basically during linking `Model_updateBoundParameters_0 `and `Model_updateBoundParameters_1` are not found!

```
/usr/bin/ld: Buildings_maint_7_0_x_Buildings_Applications_DataCenters_ChillerCooled_Equipment_Validation_NonIntegrated_08bnd.o: in function `Buildings_maint_7_0_x_Buildings_Applications_DataCenters_ChillerCooled_Equipment_Validation_NonIntegrated_updateBoundParameters':
Buildings_maint_7_0_x_Buildings_Applications_DataCenters_ChillerCooled_Equipment_Validation_NonIntegrated_08bnd.c:(.text+0xfe51): undefined reference to `Buildings_maint_7_0_x_Buildings_Applications_DataCenters_ChillerCooled_Equipment_Validation_NonIntegrated_updateBoundParameters_0'
/usr/bin/ld: Buildings_maint_7_0_x_Buildings_Applications_DataCenters_ChillerCooled_Equipment_Validation_NonIntegrated_08bnd.c:(.text+0xfe60): undefined reference to `Buildings_maint_7_0_x_Buildings_Applications_DataCenters_ChillerCooled_Equipment_Validation_NonIntegrated_updateBoundParameters_1'
clang: error: linker command failed with exit code 1 (use -v to see invocation)

```

**Comment by casella** [comment]
@adrpo, any idea when this regression was actually generated?

**Comment by casella** [comment]
@lochel, on top of the issue reported by @adrpo in the description (which I was unable to link to any specific commit), there is another issue that popped up in this [regression report](https://libraries.openmodelica.org/branches/history/master-fmi/2022-02-17%2001:51:48..2022-02-19%2001:39:48.html), which caused 287 regressions.  

All the models that stopped simulating in that report apparently did so because of this reason
```
OMSimulator: simulation/simulation_info_json.c:463: EQUATION_INFO modelInfoGetEquation(MODEL_DATA_XML *, size_t): 
Assertion `xml->equationInfo' failed.
```
see, e.g. [here](https://libraries.openmodelica.org/branches/master-fmi/OpenIPSL_release/files/OpenIPSL_release_OpenIPSL.Examples.Events.TestBreaker.sim)

I would recommend these two issues are fixed before the 1.19.0 release.

**Comment by adrpo** [comment]
Also this one:
https://libraries.openmodelica.org/branches/master-fmi/ModelicaTest_trunk/files/ModelicaTest_trunk_ModelicaTest.Blocks.FilterTests.AllOptions.err


```
/usr/bin/ld: ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_06inz.o: in function `ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_functionInitialEquations':
ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_06inz.c:(.text+0x16): undefined reference to `ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_functionInitialEquations_0'
/usr/bin/ld: ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_06inz.c:(.text+0x21): undefined reference to `ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_functionInitialEquations_1'
/usr/bin/ld: ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_06inz.c:(.text+0x2c): undefined reference to `ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_functionInitialEquations_2'
/usr/bin/ld: ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_06inz.c:(.text+0x37): undefined reference to `ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_functionInitialEquations_3'
/usr/bin/ld: ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_08bnd.o: in function `ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_updateBoundParameters':
ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_08bnd.c:(.text+0x1cec1): undefined reference to `ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_updateBoundParameters_0'
/usr/bin/ld: ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_08bnd.c:(.text+0x1ced0): undefined reference to `ModelicaTest_trunk_ModelicaTest_Blocks_FilterTests_AllOptions_updateBoundParameters_1'
```

**Comment by arun3688** [comment]
> @lochel, on top of the issue reported by @adrpo in the description (which I was unable to link to any specific commit), there is another issue that popped up in this [regression report](https://libraries.openmodelica.org/branches/history/master-fmi/2022-02-17%2001:51:48..2022-02-19%2001:39:48.html), which caused 287 regressions.
> 
> All the models that stopped simulating in that report apparently did so because of this reason
> 
> ```
> OMSimulator: simulation/simulation_info_json.c:463: EQUATION_INFO modelInfoGetEquation(MODEL_DATA_XML *, size_t): 
> Assertion `xml->equationInfo' failed.
> ```
> 
> see, e.g. [here](https://libraries.openmodelica.org/branches/master-fmi/OpenIPSL_release/files/OpenIPSL_release_OpenIPSL.Examples.Events.TestBreaker.sim)
> 
> I would recommend these two issues are fixed before the 1.19.0 release.

@casella I found the problem,  this is related with `--fmiFilter=protected` and reading the `simulation_info.json` from source folder, it seems the json file is exported to both resource and source folder.

**Comment by lochel** [comment]
Thanks @arun3688, that's a good starting point 👍

**Comment by arun3688** [comment]
@casella the regressions which failed due to assertion error 
```
OMSimulator: simulation/simulation_info_json.c:463: EQUATION_INFO modelInfoGetEquation(MODEL_DATA_XML *, size_t): 
Assertion `xml->equationInfo' failed.
```
should be fixed with this commit https://github.com/OpenModelica/OpenModelica/commit/46915e1539e67a4adfa5aefefe62dba15e31cd7d

I do not know what is causing the linking issue for `Model_updateBoundParameters_0`

**Comment by lochel** [comment]
The same issue is also reported by @dhblum: #8805

**Comment by casella** [comment]
The situation has slightly improved

![immagine](https://user-images.githubusercontent.com/4233148/162843505-8efff2ba-8d6c-4a28-bcb6-ee38f263b037.png)
![immagine](https://user-images.githubusercontent.com/4233148/162843575-5b663802-1035-4c43-96ac-34dc78c94b74.png)

but there are still a large number of regressions in those two libraries (and others). 

@arun3688, @lochel, is this just due to #8805? It would be good to fix this in the 1.19.0, to avoid large regressions with FMI export.

**Comment by casella** [comment]
This should now be fixed by #8836. Let's wait a few days until 1d7fb13 gets in the FMItest pipeline and then check the [overview-fmi](https://libraries.openmodelica.org/branches/overview-fmi.html) report to confirm. Thanks @mahge for the fix!

**Comment by adrpo** [comment]
[master-fmi 2022-04-14 01:43:05..2022-04-15 01:41:19.html](https://libraries.openmodelica.org/branches/history/master-fmi/2022-04-14%2001:43:05..2022-04-15%2001:41:19.html) 584 improved, 41 regressions; performance 79 improved, 112 regressions
